src/node_contextify.cc

```diff
@@ -289,9 +289,11 @@ void ContextifyContext::WeakCallback(
 void ContextifyContext::WeakCallbackCompileFn(
     const WeakCallbackInfo<CompileFnEntry>& data) {
   CompileFnEntry* entry = data.GetParameter();
-  entry->env->id_to_function_map[entry->id].Reset();
-  entry->env->id_to_function_map.erase(entry->id);
-  delete entry;
+  if (entry->env->compile_fn_entries.erase(entry) != 0) {
+    entry->env->id_to_function_map[entry->id].Reset();
```

Comment:
You can also remove this line because we’re using `node::Persistent` now :)

Reply:
So the idea here is that if you were to spin up lots of vm modules, none of the compile function entries would be disposed at all while the environment is alive. This code is thus attempting to clear these out during the environment execution.

Reply:
Yeah, I was just saying, the next line implies this one (it will also call `Reset()` on the Persistent, from its destructor), so it’s redundant?

Reply:
Ahh of course, thanks, I've added that now.